api_tests/src/ledgers/bitcoind_instance.ts

```diff
@@ -100,14 +93,54 @@ export class BitcoindInstance implements BitcoinInstance {
         };
     }
 
-    public async stop() {
-        this.logger.info("Stopping bitcoind instance");
+    private async findBinary(): Promise<string> {
+        const envOverride = process.env.BITCOIND_BIN;
+
+        if (envOverride) {
+            this.logger.info(
+                "Overriding bitcoind bin with BITCOIND_BIN: ",
+                envOverride
+            );
+
+            return envOverride;
+        }
+
+        const cacheDir = findCacheDir({
+            name: "bitcoin-core-0.17.0",
+            create: true,
+            thunk: true,
+        });
+        const binary = cacheDir("bitcoin-0.17.0", "bin", "bitcoind");
+
+        if (await existsAsync(binary)) {
+            return binary;
+        }
+
+        this.logger.info("Binary not found at ", binary, ", downloading ...");
+
+        await download(BitcoindInstance.downloadUrl(), cacheDir(""), {
+            decompress: true,
+            extract: true,
+        });
+
+        this.logger.info("Download completed");
+
+        return binary;
+    }
 
-        this.process.kill("SIGINT");
+    private static downloadUrl() {
+        switch (platform()) {
+            case "darwin":
+                return "https://bitcoincore.org/bin/bitcoin-core-0.17.0/bitcoin-0.17.0-osx64.tar.gz";
+            case "linux":
+                return "https://bitcoincore.org/bin/bitcoin-core-0.17.0/bitcoin-0.17.0-x86_64-linux-gnu.tar.gz";
```

Comment:
❓  So whenever we want to update to the newest version we have to check what the current version number is and change both links accordingly? Is that by design?

Reply:
An update would mean changing it in code I suppose. Once merged everybody would have the update. Correct me if I'm wrong.

Reply:
This is already the case if you look at the shell scripts that currently download the blockchain nodes.

Reply:
In fact, the whole point of this is (if you read the commit messages) that we can now in fact smoothly update these versions instead of everyone having to delete the nodes from the directory to actually trigger the re-download of the newer version.

With these scripts, they are now stored in a folder inside `node_modules` that contains the version number as-well and the nodes are reliably downloaded if we want a different version.

Reply:
Read the commit messages, they explain it it detail :)

Reply:
Do you see an alternative way of doing this that gives us stable builds?

Reply:
Do you see an alternative way of doing this that gives us stable builds?